### PR TITLE
[Enhancement](tvf) Add frontends table-valued-function

### LINKED ISSUE #20646: [Enhancement] add Frontends table-valued-function
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

This pr implement a new Metadata TVF called frontends.
The use of frontends TVF like:
select * from frontends()\G;

### Solution

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yongjinhou**
## Proposed changes

Issue Number: close #20646

This pr implement a new Metadata TVF called frontends.
The use of frontends TVF like:
select * from frontends()\G;

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by yongjinhou** [comment]
run buildall

**Review by pengxiangyu — DISMISSED** [review]
LGTM

**Review by morrySnow — COMMENTED** [review]
u need to implement this function for Nereids(new optimizer) too

**Comment by yongjinhou** [comment]
run buildall

**Comment by yongjinhou** [comment]
LGTM

**Review by pengxiangyu — DISMISSED** [review]
LGTM

**Comment by yongjinhou** [comment]
run Need_2_Approval

File: gensrc/thrift/Types.thrift [hunk 1]
```diff
@@ -687,6 +687,7 @@ enum TSortType {
 enum TMetadataType {
   ICEBERG,
   BACKENDS,
+  FRONTENDS,
```
Comment (luozenglin) [inline gensrc/thrift/Types.thrift:690]: Never add enum definition in the middle of the existing thrift definition. https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=240883637

**Comment by yongjinhou** [comment]
run buildall

File: gensrc/thrift/Types.thrift (hunk 1 above)
Comment (yongjinhou) [reply gensrc/thrift/Types.thrift]: done

**Comment by yongjinhou** [comment]
> u need to implement this function for Nereids(new optimizer) too

I will do it latter

**Comment by yongjinhou** [comment]
run p0

**Comment by yongjinhou** [comment]
run p0

**Comment by yongjinhou** [comment]
run buildall

**Review by luozenglin — APPROVED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**